### ISSUE TITLE
Transparent proxy using redirect method doesn't seem to clean itself up properly

### ISSUE THREAD

**Issue by ishland** [v2rayA/v2rayA#403]
### Version, approach of installation, operating system

1. What version of v2rayA are you using?
`1.5.6.2`

2. What approach of installation do you install V2rayA?
Using APT on Debian. 

3. What is your operating system?
Debian 11

### Description

The title says it all. 
Transparent proxy using redirect method doesn't seem to clean itself up properly. 
I have to manually disable by doing cleanup shell commands on my own.  

### Steps to reproduce

1. Enable transparent proxy with redirect method. 
2. Disable transparent proxy. 
3. Open a browser and visit any website. 

### Logs：
Some addresses are replaced by `<ipv4 address>` or `<ipv6 address>`
```
Jan 02 17:19:49 SomeHost systemd[1]: Started v2rayA Service.
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:25]  network is connected
Jan 02 17:19:49 SomeHost v2raya[8030]: [I] [pre.go:207]  init DB
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  V2RayLocationAsset is /etc/v2raya
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  V2Ray binary is
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  v2rayA working directory is /
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  v2rayA configuration directory is /etc/v2raya
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  Golang: go1.17.3
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  OS: linux
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  Arch: amd64
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  Lite: false
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  Version: 1.5.5
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [main.go:27]  Starting...
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://127.0.0.1:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv4 address>:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://[::1]:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv6 address>:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv6 address>:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv6 address>:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv6 address>:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv6 address>:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv6 address>:2017
Jan 02 17:19:49 SomeHost v2raya[8030]: [A] [index.go:131]  v2rayA is listening at http://<ipv6 address>:2017
Jan 02 17:22:51 SomeHost v2raya[8030]: [I] [io.go:425]  V2Ray 4.44.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.17.3 linux/amd64)
Jan 02 17:22:51 SomeHost v2raya[8030]: [I] [io.go:425]  A unified platform for anti-censorship.
Jan 02 17:22:51 SomeHost v2raya[8030]: [I] [io.go:425]  [Info] main/jsonem: Reading config: /etc/v2raya/config.json
Jan 02 17:22:51 SomeHost v2raya[8030]: [I] [io.go:425]  [Warning] V2Ray 4.44.0 started
Jan 02 17:24:29 SomeHost v2raya[8030]: [I] [io.go:425]  V2Ray 4.44.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.17.3 linux/amd64)
Jan 02 17:24:29 SomeHost v2raya[8030]: [I] [io.go:425]  A unified platform for anti-censorship.
Jan 02 17:24:29 SomeHost v2raya[8030]: [I] [io.go:425]  [Info] main/jsonem: Reading config: /etc/v2raya/config.json
Jan 02 17:24:30 SomeHost v2raya[8030]: [I] [io.go:425]  127.0.0.1:36088 accepted tcp:127.0.0.1:0 [api-out]
......
```

**Comment by cubercsl** [comment]
@ishland Do you mean iptables did not reset when switching transparent proxy to "off" from any other mode?

**Comment by ishland** [comment]
Yeah basically

**Comment by cubercsl** [comment]
It seem that something wrong with this...

https://github.com/v2rayA/v2rayA/blob/3e2cc0818e2b9ec55a21e6a5964f09b8f768fa67/service/core/v2ray/transparent.go#L91-L96

**Comment by cubercsl** [comment]
e7839b5e9de01f8dca66449eb2e720963a56f403

https://github.com/v2rayA/v2rayA/actions/runs/1799380228

**Comment by mzz2017** [comment]
Fixed in v1.5.7
